### PR TITLE
[SofaConstraint] Remove dependency on TetrahedronFEMForcefield

### PR DESCRIPTION
**Pull request by fredroy**
TL;DR (dev): Remove direct dependency in SofaConstraint&SofaPrecond on SofaSimpleFEM by implementing a function in TetraFEM.
TL;DR (user): Using PrecomputedConstraintCorrection with `rotations="true"` now **requires** to have a RotationFinder.

PrecomputedConstraintCorrection has a hard dependency on TetrahedronFEMForceField, to get rotations.
(hence calling a a function of TetraFEM)
It can also get rotations if it detects a component implemented RotationFinder<T>.
But TetrahedronFEMForceField is already implementing BaseRotationFinder taking a BaseMatrix*.
RotationFinder<T> inherits BaseRotationFinder, so I made TetrahedronFEMForceField implementing RotationFinder<T> instead. (which adds an other method to implement)
Finally, PrecomputedConstraintCorrection only needs RotationFinder<T> now.

IMO, 
- BaseRotationFinder implemented in TetrahedronFEM & HexahedronFEM only and used in WarpPreconditioner;
- RotationFinder<T> (implemented by nobody in the sofacode base but apparently by one external plugin @pedroperrusi) and used by PrecomputedConstraintCorrection, WarpPreconditioner and PrecomputedWarpPreconditioner;

would really need a refactoring or some re-working.
Those two were defined more or less in the same time, in 2006 (!) in the early days of SOFA.

Some cleaning in PrecomputedConstraintCorrection as well
and this PR made me wonder the never-defined preprocessor definition NEW_METHOD_UNBUILT in PrecomputedConstraintCorrection  (@ChristianDuriez ?)

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]
